### PR TITLE
 Deduplicate ways to get table rows with new get_table_rows_by_caption helper method

### PR DESCRIPTION
**Pull request by benvand**
* Add new get_table_rows_by_caption helper method
* Update existing methods to use new normalised method

### CODE REVIEW

File: features/step_definitions/common_steps.rb [hunk 1]
```diff
@@ -310,9 +310,7 @@
 end
 
 Then /^I see the '(.*)' summary table filled with:$/ do |table_heading, table|
```
Comment (benvand) [inline features/step_definitions/common_steps.rb]: Use new method here:

File: features/supplier/supplier_applies_for_an_opportunity.feature [hunk 2]
```diff
@@ -332,9 +332,9 @@ Scenario: Supplier changes their answers before submission
 
   When I click the 'View your account' link
   And I click the 'View your opportunities' link
```
Comment (benvand) [inline features/supplier/supplier_applies_for_an_opportunity.feature]: Bring in line with briefs changes from yday:
https://github.com/alphagov/digitalmarketplace-briefs-frontend/pull/63

File: features/support/helpers.rb [hunk 3]
```diff
@@ -121,3 +121,10 @@ def escape_xpath(string)
     "'#{string}'"
   end
 end
```
Comment (benvand) [inline features/support/helpers.rb]: Add new method here:

File: features/step_definitions/common_steps.rb [hunk 4]
```diff
@@ -322,16 +320,12 @@
...
-  result_table_rows = all(:xpath, result_table_rows_location)
+  result_table_rows = get_table_rows_by_caption(table_heading)
   result_table_rows.any? {|row| row.text.include? content}.should be true
 end
 
 Then /^I see that the '(.*)' summary table has (\d+)(?: or (more|fewer))? entr(?:y|ies)$/ do |table_heading, expected_number_of_rows, comparison|
-  result_table_location = "//*[@class='summary-item-heading'][normalize-space(text())=\"#{table_heading}\"]/following-sibling::*[1]"
-  result_table_rows_location = result_table_location + "/tbody/tr[@class='summary-item-row']"
-  number_of_table_rows = all(:xpath, result_table_rows_location).length
+  number_of_table_rows = get_table_rows_by_captiong(table_heading).length
```
Comment (CrystalPea) [inline features/step_definitions/common_steps.rb]: I think there might be a typo here: "get_table_rows_by_captiong" instead of "get_table_rows_by_caption"

Comment (benvand) [reply features/step_definitions/common_steps.rb]: Good catch, thanks! Updated

**Comment by benvand** [comment]
Awesome thanks! @CrystalPea @katstevens

**Pull request merged**